internal/services/queueService.go

```diff
@@ -56,31 +73,173 @@ func (qs *queueService) Listen() {
 		err := json.Unmarshal(msg.Body, &queueMessage)
 		if err != nil {
 			qs.logger.Errorf("Failed to unmarshal message: %s", err)
+			err = PublishErrorToResponseQueue(qs.channel, msg.ReplyTo, queueMessage.Type, queueMessage.MessageID, err)
+			if err != nil {
+				qs.logger.Errorf("Failed to publish error message: %s", err)
+			}
 			continue
 		}
 
-		response, err := qs.Worker.ProcessMessage(queueMessage)
-		if err != nil {
-			qs.logger.Errorf("Failed to process message: %s", err)
-			continue
+		if queueMessage.Type == constants.QueueMessageTypeTask {
```

Comment:
Instead of making such long branches. Extract logic into private functions. Llike handleTaskMessage, handleStatusMessage, etc.